script.js

```diff
@@ -0,0 +1,621 @@
+  let timesDifferenceIsMinusOne = 0;
+  // if there is a flush, check for straight from 10 to ace
+  if (isFlush() === true) {
+    // check if player has a straight from 10 to ace
+    for (let i = 0; i < (playerHand.length - 2); i += 1) {
+      if (playerHand[i].rank - playerHand[i + 1].rank === 1) {
+        timesDifferenceIsMinusOne += 1;
+      }
+    }
+    if (timesDifferenceIsMinusOne === 3 && playerHand[4].rank === 1 && playerHand[0].rank === 13) {
+      check10ToAce = true;
+    }
+  }
+
+  return check10ToAce;
+};
+// returns true if there is a full house in the player's hand
+const isFullHouse = () => {
+  let checkFullHouse = false;
+  if (numOf3OfAKind === 1 && numOfPairs === 1) {
+    checkFullHouse = true;
+  }
+  return checkFullHouse;
+};
+// returns true if there is a card that is Jack or higher in the player's hand
+const isJackOrHigher = () => {
+  let checkJackOrHigher = false;
+  // check every card in player's hand for jack,queen,king or ace
+  for (let i = 0; i < playerHand.length; i += 1) {
+    if (playerHand[i].rank > 10 || playerHand[i].rank === 1) {
+      checkJackOrHigher = true;
+    }
+  }
+  return checkJackOrHigher;
+};
+// returns number of points based on player's hand
+const calcHandScore = () => {
+  if (isRoyalFlush() === true) { // royal flush
+    handScore = 10;
+  } else if (isStraight() === true && isFlush() === true) { // straight flush
+    handScore = 9;
+  } else if (numOf4OfAKind === 1) { // 4 of a kind
+    handScore = 8;
+  } else if (isFullHouse() === true) { // full house
+    handScore = 7;
+  } else if (isFlush() === true) { // flush
+    handScore = 6;
+  } else if (isStraight() === true) { // straight
+    handScore = 5;
+  } else if (numOf3OfAKind === 1) { // 3 of a kind
+    handScore = 4;
+  } else if (numOfPairs === 2) { // 2 pairs
+    handScore = 3;
+  } else if (numOfPairs === 1) { // 1 pair
+    handScore = 2;
+  } else if (isJackOrHigher() === true) {
+    handScore = 1;
+  } else {
+    handScore = 0;
+  }
+};
+
+// returns points won for the round
+const calcPointsWon = () => playerBidPoints * handScore;
+// add points to player's total points based on bid points and hand score
+const addPoints = () => {
+  playerTotalPoints += calcPointsWon();
+};
+
+// Game initialization =============================================
+// to initilize game
+const initGame = () => {
+  // make and store a shuffled deck
+  deck = shuffleCards(makeDeck());
+
+  // initialize starting elements
+  createStartingElements();
+
+  // initialize playerHandContainer functionality
+  gameDisplayContainer.appendChild(playerHandContainer);
+
+  // initialize gameInfo functionality
+  displayGameInfo('Welcome! Please submit your bid points to start playing.');
+  gameDisplayContainer.appendChild(gameInfo);
+
+  // initialize dealButton functionality
+  dealButton.setAttribute('id', 'deal-button');
+  dealButton.innerText = 'deal cards';
+  dealButton.addEventListener('click', () => {
```

Comment:
Would recommend separating the UI logic from the game logic. The reason for this is to make it easier for debugging. If it's a UI issue, we can focus on the UI functions. If it's a game logic issue, we can focus on the game logic functions. You've mostly adhered to this above in the helper functions, but can go further by trying to separate UI from game logic in the `initGame` function.

E.g. when adding event listeners, make the callback function (which contains mostly game logic) a helper function.

Reply:
Since these callback helper functions will perform different tasks, I am unsure how to name this helper function. Can I just name it as `handle`+ `variable name that uses the addEventListener method`? (Eg. Naming callback function at line 480 as `handleDealButton`).

Reply:
`handle` + `<elem_name>` + `<event_type>`, e.g. `handleSwapButtonClick`